src/main/features/response/components/payment-intention/payment-plan.ts

```diff
@@ -0,0 +1,107 @@
+  }
+
+  const { totalAmount, instalmentAmount, paymentSchedule } = model
+  if (totalAmount && instalmentAmount && paymentSchedule) {
+    return createPaymentPlan(totalAmount, instalmentAmount, mapFrequencyInWeeks(paymentSchedule)).getPaymentLength()
+  }
+
+  return undefined
+}
+
+export class PaymentPlanPage {
+  constructor (private admissionType: string) {}
+
+  buildRouter (path: string = ''): express.Router {
+    const stateGuardRequestHandler: express.RequestHandler = GuardFactory.create((res: express.Response): boolean => {
+      const draft: Draft<ResponseDraft> = res.locals.responseDraft
+
+      return draft.document[this.admissionType]
+        && draft.document[this.admissionType].paymentOption
+        && draft.document[this.admissionType].paymentOption.isOfType(DefendantPaymentType.INSTALMENTS)
+    }, (req: express.Request, res: express.Response): void => {
+      const claim: Claim = res.locals.claim
+
+      res.redirect(Paths.taskListPage.evaluateUri({ externalId: claim.externalId }))
+    })
+
+    return express.Router()
+      .get(path + PaymentIntentionPaths.paymentPlanPage.uri,
+        FeatureToggleGuard.featureEnabledGuard(this.admissionType),
+        stateGuardRequestHandler,
+        ErrorHandling.apply(async (req: express.Request, res: express.Response) => {
+          const draft: Draft<ResponseDraft> = res.locals.responseDraft
+          this.renderView(new Form(draft.document[this.admissionType].paymentPlan), res)
+        }))
+      .post(path + PaymentIntentionPaths.paymentPlanPage.uri,
+        FeatureToggleGuard.featureEnabledGuard(this.admissionType),
+        stateGuardRequestHandler,
+        FormValidator.requestHandler(DefendantPaymentPlan, DefendantPaymentPlan.fromObject, undefined, ['calculatePaymentPlan']),
+        ErrorHandling.apply(
+          async (req: express.Request, res: express.Response): Promise<void> => {
+            const form: Form<DefendantPaymentPlan> = req.body
+            if (form.hasErrors() || _.get(req, 'body.action.calculatePaymentPlan')) {
+              this.renderView(form, res)
+            } else {
+              const draft: Draft<ResponseDraft> = res.locals.responseDraft
+              const user: User = res.locals.user
+
+              draft.document[this.admissionType].paymentPlan = form.model
+              await new DraftService().save(draft, user.bearerToken)
+
+              const { externalId } = req.params
+              res.redirect(Paths.taskListPage.evaluateUri({ externalId: externalId }))
+            }
+          }))
+  }
+
+  renderView (form: Form<DefendantPaymentPlan>, res: express.Response): void {
+    const claim: Claim = res.locals.claim
+    const draft: Draft<ResponseDraft> = res.locals.responseDraft
+
+    res.render('response/components/payment-intention/payment-plan.njk', {
+      form: form,
+      paymentLength: calculatePaymentPlanLength(form.model),
+      monthlyIncome: _.get(draft, 'document.statementOfMeans.monthlyIncome', 0),
```

Comment:
not sure if i like this, there's no type safety, I assume you're trying to avoid undefined checks, but I think having the types is better than doing this.
REC

Reply:
Seems like something we accepted other time. I think we should change it to use more type safe approach.

Reply:
Sadly it never worked due to the fact `document.statementOfMeans.monthlyIncome` does not hold any numeric value, it is just data object that holds data necessary for total calculation.

Since it requires rework it will be done as separate story / PR.